develop/blockchains/deployment/generate-chain-specs.md

```diff
@@ -1,4 +1,136 @@
+
+- `name` - the human-readable name for the network
+- `id` - the machine-readable identifier for the network
+- `chainType` - the type of chain to start (refer to [ChainType](https://paritytech.github.io/polkadot-sdk/master/sc_chain_spec/enum.ChainType.html){target=\_blank} for more details)
+- `bootNodes` - a list of multiaddresses belonging to the chain's boot nodes
+- `telemetryEndpoints` - an optional list of multiaddresses for telemetry endpoints, with verbosity levels ranging from 0 to 9 (0 being the lowest verbosity)
+- `protocolId` - the optional protocol identifier for the network
+- `forkId` - an optional fork ID that should typically be left empty; it can be used to signal a fork at the network level when two chains share the same genesis hash
+- `properties` - custom properties provided as a key-value JSON object
+- `codeSubstitutes` - an optional mapping of block numbers to Wasm code
+- `genesis` - the genesis configuration for the chain
+
+For example, the following JSON shows a basic chain specification file:
+
+```json
+--8<-- "code/develop/blockchains/deployment/generate-chain-specs/basic-chain-specs.json"
+``` 
+
+
+## Creating a Custom Chain Specification
+
+To create a custom chain specification, you can use the [`chain-spec-builder`](https://paritytech.github.io/polkadot-sdk/master/staging_chain_spec_builder/index.html){target=\_blank} tool. This is a CLI tool that is used to generate chain specifications from the runtime of a node. To install the tool, run the following command:
+
+```bash
+cargo install staging-chain-spec-builder
+```
+
+To verify the installation, run the following:
+
+```bash
+chain-spec-builder --help
+```
+
+### Plain Chain Specifications
+
+To create a plain chain specification, you can use the following utility within your project:
+
+```bash
+chain-spec-builder create -r <RUNTIME_WASM_PATH> <INSERT_COMMAND> 
+```
+
+!note
+    Before running the command, ensure that the runtime has been compiled and is available at the specified path.
+
+Ensure to replace `<RUNTIME_WASM_PATH>` with the path to the runtime Wasm file and `<INSERT_COMMAND>` with the command to insert the runtime into the chain specification. The available commands are:
+
+- `patch` - overwrites the runtime's default genesis config with the provided patch. You can check the following [patch file](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/utils/chain-spec-builder/tests/input/patch.json){target=\_blank} as a reference
+- `full` - build the genesis config for runtime using the JSON file. No defaults will be used. As a reference, you can check the following [full file](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/utils/chain-spec-builder/tests/input/full.json){target=\_blank}
```

Comment:
Maybe better use frozen the links? (I mean instead of master use some hard-coded commit sha?)
Not sure if you have some agreed policy about that.

Reply:
Used version tag as follows: https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable1985-rc1/substrate/bin/utils/chain-spec-builder/tests/input/full.json